### PR TITLE
Sign compare warning fixes batch 1 fix2

### PR DESCRIPTION
**Pull request by tg-at-google**
Not ready for merge just yet, 35 files changes expected. Please await these changes.

### CODE REVIEW

**Comment by tg-at-google** [comment]
@joker-eph The mlir related changes are just copies of what is currently at tensorflow/tensorflow:master, i.e. they are actually not changes at all.

**Comment by gbaned** [comment]
@tg-at-google Can you please resolve conflicts? Thanks!

**Review by tg-at-google — COMMENTED** [review]
This represents completion of my review of this set of warning resolutions. 
Again, please note that the mlir changes are just mirror of tensorflow/tensorflow:master and are thus not changes at all. 

@mihaimaruseac @joker-eph

File: tensorflow/core/kernels/quantization_utils.h [hunk 1]
```diff
@@ -268,7 +268,7 @@ inline void RequantizeManyInNewRangeReference(const qint32* input, int64 count,
   // that could be easily adapted for a SIMD implementation. It should also be
   // possible to perform all the calculations in 32-bit rather than 64, but
   // that's not been implemented yet.
-  for (size_t index = 0; index < count; ++index) {
+  for (size_t index = 0; static_cast<tensorflow::int64>(index) < count; ++index) {
```
Comment (mihaimaruseac) [inline tensorflow/core/kernels/quantization_utils.h:271]: Let's change type of `index` to `int64` instead.

File: tensorflow/lite/toco/graph_transformations/resolve_batch_normalization.cc [hunk 2]
```diff
@@ -126,11 +126,11 @@ ::tensorflow::Status ResolveBatchNormalization::Run(Model* model,
...
-  CHECK(mul_float_data.size() == buffer_size);
-  CHECK(add_float_data.size() == buffer_size);
-  CHECK(mean_float_data.size() == buffer_size);
-  CHECK(multiplier_float_data.size() == buffer_size);
-  CHECK(offset_float_data.size() == buffer_size);
+  CHECK(static_cast<int>(mul_float_data.size()) == buffer_size);
+  CHECK(static_cast<int>(add_float_data.size()) == buffer_size);
+  CHECK(static_cast<int>(mean_float_data.size()) == buffer_size);
+  CHECK(static_cast<int>(multiplier_float_data.size()) == buffer_size);
+  CHECK(static_cast<int>(offset_float_data.size()) == buffer_size);
```
Comment (mihaimaruseac) [inline tensorflow/lite/toco/graph_transformations/resolve_batch_normalization.cc:133]: Let's change `buffer_size`'s type instead

File: tensorflow/lite/toco/graph_transformations/unpartition_embedding_lookup.cc [hunk 3]
```diff
@@ -55,12 +55,12 @@ ::tensorflow::Status UnpartitionEmbeddingLookup::Run(Model* model,
   auto* stitch_op = static_cast<DynamicStitchOperator*>(op_it->get());
 
   // Split up the DynamicStitch inputs into the indices and data.
-  std::vector<std::string> stitch_indices_inputs;
-  std::vector<std::string> stitch_data_inputs;
-  for (size_t i = 0; i < stitch_op->num_partitions; ++i) {
+  std::vector<string> stitch_indices_inputs;
+  std::vector<string> stitch_data_inputs;
```
Comment (mihaimaruseac) [inline tensorflow/lite/toco/graph_transformations/unpartition_embedding_lookup.cc:59]: We should keep `std::string` here

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]
Forgot the other 2 items

**Comment by tg-at-google** [comment]
> Forgot the other 2 items

Hey, yeah, wasn't done yet. 😅

**Comment by tg-at-google** [comment]
> > Forgot the other 2 items
> 
> Hey, yeah, wasn't done yet. 😅

Alright @mihaimaruseac  good to go.

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**